cmd/collector/app/http_handler_test.go

```diff
@@ -43,6 +43,14 @@ type mockJaegerHandler struct {
 }
 
 func (p *mockJaegerHandler) SubmitBatches(ctx tchanThrift.Context, batches []*jaeger.Batch) ([]*jaeger.BatchSubmitResponse, error) {
+	return p.submitBatches(ctx, batches)
+}
+
+func (p *mockJaegerHandler) SubmitHTTPBatches(ctx tchanThrift.Context, batches []*jaeger.Batch) ([]*jaeger.BatchSubmitResponse, error) {
```

Comment:
instead of exploding the handler APIs with SubmitHTTPxxx, you pass the endpoint name via ctx (I assume it has ctx.WithValue() similar to Go Ctx)

Reply:
I've thought of go context to be used for data that changes per request (for e.g. timeout values, etc); In this particular use case, it seems that the values being passed from the would never change per request, and that a the API change is more readable. 
Is the philosophy behind Go's context very different from Java's ThreadLocals?

Reply:
Hmmm, doesn't seem like `WithValue` or alike is available for `thrift.Context`. Talked to Prithvi about this and prefer to get http handler a separate method.

Reply:
These interfaces that accept thrift.Context exist because they are passed to tchannel servers:
```
server.Register(jc.NewTChanCollectorServer(jaegerBatchesHandler))
server.Register(zc.NewTChanZipkinCollectorServer(zipkinSpansHandler))
```
So I really don't like extending them with unrelated HTTP method. I think it is time to do a bit of refactoring. We have three layers:
  * transport handlers, e.g. grpc / http / tchannel
  * representation handlers: jaegerBatchesHandler, zipkinSpansHandler (which also double as tchannel handlers, an abstraction leak)
  * processor that only takes model spans: `ProcessSpans(spans, format)`

I recommend we clearly de-couple jaegerBatchesHandler and zipkinSpansHandler from tchannel. Then representation handlers can take an Options struct, e.g. jaegerBatchesHandler:
```go
type JaegerBatchesHandler interface {
	// SubmitBatches records a batch of spans in Jaeger Thrift format
	SubmitBatches(batches []*jaeger.Batch, options SubmitBatchesOptions) ([]bool, error)
}

type SubmitBatchesOptions {
    InboundProtocol  string // http, tchannel, etc.
}
```

Similarly, refactor SpanProcessor from
```
func (sp *spanProcessor) ProcessSpans(mSpans []*model.Span, spanFormat string) ([]bool, error)
```

to 

```go
func (sp *spanProcessor) ProcessSpans(mSpans []*model.Span, options ProcessSpansOptions) ([]bool, error)

type ProcessSpansOptions struct {
    InboundProtocol  string // http, tchannel, etc.
    SpanFormat  string
}
```